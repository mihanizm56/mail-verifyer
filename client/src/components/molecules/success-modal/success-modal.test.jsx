import React, { createElement } from "react";
import { shallow } from "enzyme";
import { SuccessModal } from ".";

describe("test SuccessModal", () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("render snap with all props", () => {
			const props = {
				handleButtonClick: jest.fn(),
				successText: "success text",
				buttonClickRedirectsTo: "/test",
			};
			const component = shallow(<SuccessModal {...props} />);

			expect(component).toMatchSnapshot();
		});
		it("render OpenSansText component", () => {
			const props = {
				handleButtonClick: jest.fn(),
				successText: "success text",
				buttonClickRedirectsTo: "/test",
			};
			const component = shallow(<SuccessModal {...props} />);
			const OpenSansText = component.find("OpenSansTextMock");

			expect(OpenSansText.length).toEqual(1);
		});
		it("render LinkButton component", () => {
			const props = {
				handleButtonClick: jest.fn(),
				successText: "success text",
				buttonClickRedirectsTo: "/test",
			};
			const component = shallow(<SuccessModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");

			expect(LinkButton.length).toEqual(1);
		});
	});
	describe("test props contract", () => {
		it("set text prop to the OpenSansText", () => {
			const props = {
				handleButtonClick: jest.fn(),
				successText: "success text",
				buttonClickRedirectsTo: "/test",
			};
			const component = shallow(<SuccessModal {...props} />);
			const OpenSansText = component.find("OpenSansTextMock");
			const textProp = OpenSansText.prop("text");

			expect(textProp).toEqual(props.successText);
		});
		it("set to (path) prop to the LinkButton", () => {
			const props = {
				handleButtonClick: jest.fn(),
				successText: "success text",
				buttonClickRedirectsTo: "/test",
			};
			const component = shallow(<SuccessModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");
			const pathProp = LinkButton.prop("to");

			expect(pathProp).toEqual(props.buttonClickRedirectsTo);
		});
		it("set text prop to the LinkButton", () => {
			const props = {
				handleButtonClick: jest.fn(),
				successText: "success text",
				buttonClickRedirectsTo: "/test",
			};
			const component = shallow(<SuccessModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");
			const textProp = LinkButton.prop("text");

			expect(textProp).toEqual("buttons.send-new");
		});
		it("set handleButtonClick prop to the LinkButton", () => {
			const props = {
				handleButtonClick: jest.fn(),
				successText: "success text",
				buttonClickRedirectsTo: "/test",
			};
			const component = shallow(<SuccessModal {...props} />);
			const LinkButton = component.find("LinkButtonMock");
			const handleButtonClick = LinkButton.prop("handleButtonClick");

			expect(handleButtonClick).toEqual(props.handleButtonClick);
		});
	});
});
