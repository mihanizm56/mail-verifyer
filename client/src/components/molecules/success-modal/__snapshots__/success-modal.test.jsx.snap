// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SuccessModal test render contract render snap with all props 1`] = `
<div
  className="success-modal"
>
  <div
    className="success-modal__text"
  >
    <OpenSansTextMock
      text="success text"
    />
  </div>
  <div
    className="success-modal__button"
  >
    <LinkButtonMock
      handleButtonClick={[MockFunction]}
      text="buttons.send-new"
      to="/test"
    />
  </div>
</div>
`;
